### PR TITLE
CSD-1567 Replaced deprecated HttpDownload with HttpGet and added time…

### PR DESCRIPTION
**Pull request by yeunga**
…outs to avoid HttpGet being 'hung'.

Asked Ling to try out artifact-diff-si and artifact-download-si on sciproc2-01 using HST collection. artifact-diff-si completed without being 'hung' and artifact-download-si uploaded more than 250 files to storage inventory without being 'hung' as well.

### CODE REVIEW

File: caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactValidator.java [hunk 1]
```diff
@@ -613,8 +613,11 @@ private ArtifactMetadata getMetadata(String observationID, Artifact artifact, Da
         queryString.append(URLEncoder.encode(adql, "UTF-8"));
         URL url = new URL(baseURL.toString() + "?" + queryString.toString());
         ResultReader resultReader = new ResultReader(artifactStore);
-        HttpDownload get = new HttpDownload(url, resultReader);
+        HttpGet get = new HttpGet(url, resultReader);
+        get.setConnectionTimeout(60000);
+        get.setReadTimeout(60000);
         try {
+            get.prepare();
```
Comment (pdowler) [inline caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactValidator.java:620]: You don't normally need to call prepare() and run(). The latter calls prepare(), rteads the response , and captures all exceptions. The usual pattern is either this:
```
try {
   get.prepare();
   InputStream istream = get.getInputStream();
   ...
} catch ...
```
or this:
```
get.run();
if (get.Throwable() != null) {
   ...
```

The explicit prepare/getInputStream is recent and allows for better exception handling and simpler reading, but I would not bother changing style in caom2-artifact-sync at this point.

File: caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/DownloadArtifactFiles.java [hunk 2]
```diff
@@ -302,8 +302,11 @@ public ArtifactDownloadResult call() throws Exception {
                 URL url = caomArtifactResolver.getURL(artifactURI);
 
                 OutputStream out = new ByteArrayOutputStream();
-                HttpDownload head = new HttpDownload(url, out);
+                HttpGet head = new HttpGet(url, out);
+                head.setConnectionTimeout(60000);
+                head.setReadTimeout(60000);
                 head.setHeadOnly(true);
+                head.prepare();
```
Comment (pdowler) [inline caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/DownloadArtifactFiles.java:309]: as above: don't need explicit prepare()

**Review by pdowler — CHANGES_REQUESTED** [review]

File: caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactValidator.java (hunk 1 above)
Comment (yeunga) [reply caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/ArtifactValidator.java]: Removed prepare().

File: caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/DownloadArtifactFiles.java (hunk 2 above)
Comment (yeunga) [reply caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/DownloadArtifactFiles.java]: Removed prepare().

**Review by pdowler — APPROVED** [review]

**Pull request merged**